src/main/java/com/salesforce/dataloader/action/visitor/BulkQueryVisitor.java

```diff
@@ -71,25 +71,38 @@ protected int executeQuery(String soql) throws AsyncApiException, OperationExcep
         } catch (final UnsupportedEncodingException e) {
             throw new ExtractException(e);
         }
-        jobUtil.closeJob();
-        final BatchInfo b = jobUtil.getBatches().getBatchInfo()[0];
-        if (b.getState() == BatchStateEnum.Failed) throw new ExtractException("Batch failed: " + b.getStateMessage());
-        this.batch = b;
-        return b.getNumberRecordsProcessed();
+        jobUtil.awaitCompletionAndCloseJob();
+        
+        int recordsProcessed = 0;
+        final BatchInfo[] ba = jobUtil.getBatches().getBatchInfo();
```

Comment:
Minor, let's call it something meaning, like`batchInfos`

Reply:
done.